homeassistant/components/sensor/zoneminder.py

```diff
@@ -104,13 +121,22 @@ def state(self):
 
     def update(self):
         """Update the sensor."""
-        archived_filter = '/Archived:0'
+        date_filter = '1%20{}'.format(self._type)
+        if self._type == 'all':
+            # The consoleEvents API uses DATE_SUB, so give it
+            # something large
+            date_filter = '100%20year'
+
+        archived_filter = '/Archived=:0'
         if self._include_archived:
             archived_filter = ''
 
         event = zoneminder.get_state(
-            'api/events/index/MonitorId:%i%s.json' % (self._monitor_id,
-                                                      archived_filter)
+            'api/events/consoleEvents/{}{}.json'.format(date_filter,
+                                                        archived_filter)
         )
 
-        self._state = event['pagination']['count']
+        try:
+            self._state = event['results'][str(self._monitor_id)]
+        except (TypeError, KeyError):
+            self._state = '0'
```

Comment:
Unknown state should be returned as `None`.

Reply:
It isn't necessarily unknown; zoneminder will return an empty set if no events matched the query. Since the state contains the number of events it seems returning '0' is appropriate.

Reply:
Is there no such thing as a query actually returning zero events? We should distinguish between that case and the case where the query doesn't match anything or no return value due to error.

Reply:
Not that I have been able to determine. The zoneminder API docs are a little thin.

Reply:
Ok, then it's ok I think.